packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -681,16 +814,103 @@ export class PipelineEditor extends React.Component<
               }
             });
           }
-        } else {
-          // in this case, pipeline version is current
-          this.canvasController.setPipelineFlow(pipelineJson);
         }
       }
       this.setState({ emptyPipeline: Utils.isEmptyPipeline(pipelineJson) });
+      this.canvasController.setPipelineFlow(pipelineJson);
+      this.validateAllNodes();
     });
   }
 
+  /**
+   * Adds an error decoration if a node has any invalid properties.
+   *
+   * @param node - canvas node object to validate
+   *
+   * @returns true if the node is valid.
+   */
+  validateNode(node: any): boolean {
+    node.app_data.invalidNodeError = this.invalidProperties(node);
+    if (node.app_data.invalidNodeError != null) {
+      this.canvasController.setNodeDecorations(node.id, [
+        {
+          id: 'error',
+          image: IconUtil.encode(errorIcon),
+          outline: false,
+          class_name: 'elyra-canvasErrorIcon',
+          position: 'topLeft',
+          x_pos: 20,
+          y_pos: 3
+        }
+      ]);
+      const pipelineId = this.canvasController.getPrimaryPipelineId();
+      const stylePipelineObj: any = {};
+      stylePipelineObj[pipelineId] = [node.id];
+      const styleSpec = {
+        body: { default: 'stroke: #d32f2f;' },
+        label: { default: 'fill: #d32f2f;' }
```

Comment:
would it be possible to set these to `var(--jp-error-color1)`?
also when the node is selected there is still the thinker blue stroke around it. would it be possible to set that to red as well?

Reply:
Done